dashboard/test/controllers/api/v1/users_controller_test.rb

```diff
@@ -87,6 +87,22 @@ class Api::V1::UsersControllerTest < ActionController::TestCase
     refute @user.show_progress_table_v2
   end
 
+  test 'a post request to show_progress_table_v2 updates appropriate timestamp' do
+    sign_in(@user)
+    assert_nil @user.progress_table_v2_timestamp
+    post :post_show_progress_table_v2, params: {user_id: 'me', show_progress_table_v2: true}
+    assert_response :success
+    @user.reload
+    assert @user.progress_table_v2_timestamp
+    assert_nil @user.progress_table_v1_timestamp
+
+    post :post_show_progress_table_v2, params: {user_id: 'me', show_progress_table_v2: false}
+    assert_response :success
+    @user.reload
+    assert @user.progress_table_v2_timestamp
+    assert @user.progress_table_v1_timestamp
```

Comment:
Would it be useful to also check the v1 timestamp is after the v2 timestamp, or are these timestamps at a low enough granularity that they might often be equal if you execute them one after the other like this?

Reply:
I think checking that they are set is enough and I would be worried about another race condition or similar.